addons/mail/static/tests/chat_window/chat_window.test.js

```diff
@@ -1018,3 +1018,44 @@ test("mark as read when opening chat window", async () => {
     await contains(".o-mail-ChatWindow .o-mail-ChatWindow-header", { text: "bob" });
     await contains(".o-mail-ChatWindow-counter", { count: 0 });
 });
+
+test("chat window: notes on the record should not be squashed", async () => {
+    const pyEnv = await startServer();
+    const partnerId = pyEnv["res.partner"].create({ name: "TestPartner" });
+    const messageId = pyEnv["mail.message"].create([
+        {
+            model: "res.partner",
+            body: "Test Message",
+            author_id: partnerId,
+            needaction: true,
+            res_id: partnerId,
+        },
+        {
+            model: "res.partner",
+            body: "Hello",
+            author_id: serverState.partnerId,
+            needaction: true,
+            res_id: partnerId,
+            is_note: true,
+        },
+        {
+            model: "res.partner",
+            body: "World!",
+            author_id: serverState.partnerId,
+            needaction: true,
+            res_id: partnerId,
+            is_note: true,
+        },
+    ]);
+    pyEnv["mail.notification"].create({
+        mail_message_id: messageId[0],
+        notification_status: "sent",
+        notification_type: "inbox",
+        res_partner_id: serverState.partnerId,
+    });
+    await start();
+    await click(".o_menu_systray i[aria-label='Messages']");
+    await click(".o-mail-NotificationItem");
+    await contains(".o-mail-Message:not(.o-squashed)", { text: "Hello" });
+    await contains(".o-mail-Message:not(.o-squashed)", { text: "World!" });
```

Comment:
- Asserting `:not(.o-squashed)` is not reliable without a similar positive assertion. Should add bubble messages that are squashed, so that you assert `.o-mail-Message.o-squashed` positively, to contrast well with the negative assertion.

- Also this is best to assert something that the user sees rather than a technical classname. It's alright here because it's functionally easy to grasp what a squashed message mean, but we could be more concrete by asserting avatar / user name that is part of header when a message is not squashed.